rdmo/accounts/models.py

```diff
@@ -123,6 +130,64 @@ class Meta:
     def __str__(self):
         return self.user.username
 
+    @classmethod
+    def create_consent(cls, user, session=None) -> bool:
+        obj, _created = cls.objects.update_or_create(user=user, defaults={"consent": True})
+
+        # Validate consent before storing in session
+        if obj.is_consent_valid():
+            if session:
+                session[CONSENT_SESSION_KEY] = True
+            return True
+
+        obj.delete()  # Remove when consent is outdated
+        return False
+
+
+    @classmethod
```

Comment:
I actually likes it better when this was in `utils`. (I just forgot to pull and looked at an older version of the branch.)

Reply:
Ok after looking at it some more, it makes sense together with the other methods.

Reply:
I used the "Fat Models" approach for this and put everything relevant on the model..